packages/blob-storage/internal/indexer/indexer.go

```diff
@@ -0,0 +1,94 @@
+package indexer
+
+import (
+	"context"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// Indexer struct holds the configuration and state for the Ethereum chain listener.
+type Indexer struct {
+	rpcURL            string
+	beaconURL         string
+	networkName       string
+	contractAddress   common.Address
+	eventHash         common.Hash
+	startHeight       *big.Int
+	eventCallbackFunc func(string, string, string, types.Log)
+}
+
+// NewIndexer creates a new Indexer instance.
+func NewIndexer(rpcURL, beaconURL, networkName, contractAddress string, pastEvents bool, startBlockNumber *big.Int) *Indexer {
+	client, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		log.Fatal("Failed to connect to the Ethereum client:", err)
+	}
+
+	if pastEvents == false {
+		header, err := client.HeaderByNumber(context.Background(), nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+		startBlockNumber = header.Number
+	}
+
+	slog.Info(client.Client())
+
+	return &Indexer{
+		rpcURL:          rpcURL,
+		beaconURL:       beaconURL,
+		networkName:     networkName,
+		contractAddress: common.HexToAddress(contractAddress),
+		startHeight:     startBlockNumber,
+	}
+}
+
+// SubscribeEvent subscribes to the specified event.
+func (indexer *Indexer) SubscribeEvent(eventHash string, callbackFunc func(string, string, string, types.Log)) {
```

Comment:
callbackFunc should be of type CallbackFunc, which is defined but not used here. callbackFunc is also a non descriptive name - "onEvent" or something is better, to show when the function is used